File: src/_03_BehavioralPattern/_03_05_Mediator/java/after/CleaningService.java
package _03_BehavioralPattern._03_05_Mediator.java.after;

public class CleaningService {
  
  private FrontDesk frontDesk;

  public CleaningService(FrontDesk frontDesk) {
    this.frontDesk = frontDesk;
  }

  public void getTowers(Integer guestId, int numberOfTowers) {
    String roomNumber = this.frontDesk.getRoomNumberFor(guestId);
    System.out.println("towers " + numberOfTowers + " to " + roomNumber);
  }
  
}

File: src/_03_BehavioralPattern/_03_05_Mediator/CleaningService.js
/**
 * @typedef { import("./FrontDesk").default } FrontDesk
 */

export default class CleaningService {
  /** @type { FrontDesk } */
  _frontDesk;

  /** @param { FrontDesk } frontDesk */
  constructor(frontDesk) {
    this._frontDesk = frontDesk;
  }

  /**
   * @param { number } guestId 
   * @param { number } numberOfTowers 
   */
  getTowers(guestId, numberOfTowers) {
    const roomNumber = this._frontDesk.getRoomNumber(guestId);
    console.log(`${roomNumber} 에 수건 ${numberOfTowers} 개 전달`);
  }
}